internal/controller/storageclient_controller.go

```diff
@@ -697,15 +721,48 @@ func removeStorageClaimAsOwner(obj client.Object) {
 	}
 }
 
-func (r *storageClientReconcile) reconcileResource(obj client.Object, rawObject []byte) error {
-	objectMeta := &metav1.PartialObjectMetadata{}
-	if err := json.Unmarshal(rawObject, objectMeta); err != nil {
-		return err
+func (r *storageClientReconcile) reconcileResourcesByGVK(
+	obj client.Object,
```

Comment:
Object is a bad name because we don't use these as 'objects', more as types

Reply:
changed to `objectType`